at_client_mobile/README.md

```diff
@@ -1,8 +1,121 @@
 # at_client_mobile
-This SDK provides the essential methods for building an app using the @protocol
-with useful device specific features for iOS and Android applications.
\ No newline at end of file
+
+### Introduction
+
+A Flutter extension to the at_client library which adds support for mobile, desktop and IoT devices.
+
+SDK that provides the essential methods for building an app using [The @protocol](https://atsign.com). You may also want to look at [at_client](https://pub.dev/packages/at_client).
+
+**at_client_mobile** package is written in Dart, supports Flutter and follows the
+@platform's decentralized, edge computing model with the following features: 
+- Cryptographic control of data access through personal data stores
+- No application backend needed
+- End to end encryption where only the data owner has the keys
+- Private and surveillance free connectivity
+
+We call giving people control of access to their data "*flipping the internet*".
+
+## Get Started
+
+Initially to get a basic overview of the SDK, you must read the [atsign docs](https://atsign.dev/docs/overview/).
+
+> To use this package you must be having a basic setup, Follow here to [get started](https://atsign.dev/docs/get-started/setup-your-env/).
+
+Check how to use this package in the [at_client_mobile installation tab](https://pub.dev/packages/at_client_mobile/install).
+
+## Usage
+
+- Get `KeyChainManager` instance to manage your keys while switching between atsigns.
+
+```dart
+import 'package:at_client_mobile/at_client_mobile.dart';
+
+static final KeyChainManager _keyChainManager = KeyChainManager.getInstance();
+
+/// Fetch atsign from the keychain manager
+String atSign = await _keyChainManager.getAtSign();
+```
+
+- Delete atsign from the keychain manager
+
+```dart
+await _keyChainManager.deleteAtSignFromKeychain(atsign);
+```
+
+- Fetch List of atsigns from the keychain manager.
+
+```dart
+List<String>? atSignsList = await _keyChainManager.getAtSignListFromKeychain();
+```
+
+- Make an atsign primary in device storage.
+
+```dart
+AtClientManager.getInstance().setCurrentAtSign(atsign, AppConstants.appNamespace, AtClientPreference());
+
+bool isAtsignSetPrimary = await _keyChainManager.makeAtSignPrimary(atsign);
+
+print(isAtsignSetPrimary); // Prints true if set primary.
+```
+
+- Get atsign status from device storage.
+
+```dart
+Map<String, bool?> atSignsWithState = await _keyChainManager.getAtsignsWithStatus();
+
+print(atSignsWithState); // Prints a map of atsigns with their status.
+
+/// Output:
+/// {
+///   "@atsign1": true, // @atsign1 is set as primary
+///   "@atsign2": false,
+///   "@atsign3": false
+/// }
+```
+
+- Reset atsigns from device storage.
+
+```dart
+for (String atsign in atSignsList) {
+    await _keyChainManager.resetAtSignFromKeychain(atsign);
+}
+```
+
+- Get `AtClientService` instance to manage your atsigns.
+
+```dart
+Map<String?, AtClientService> atClientServiceMap = {};
+
+OnboardingWidgetService().onboarding(
```

Comment:
This piece of code does not give details of any context. Like where it came from.